drivers/SmartThings/wemo/src/subscription_server.lua

```diff
@@ -29,11 +29,19 @@ SubscriptionServer.mt = {
 }
 
 function SubscriptionServer:subscribe(device)
-  self.ctrl_tx:send(ControlMessage.Subscribe(device))
+  if device ~= nil then
+    self.ctrl_tx:send(ControlMessage.Subscribe(device))
+  else
+    log.warn("serve| invalid device, cannot subscribe")
+  end
 end
 
 function SubscriptionServer:unsubscribe(device)
-  self.ctrl_tx:send(ControlMessage.Unsubscribe(device))
+  if device == nil then
```

Comment:
Is this intended to be `device==nil` or should it be `device~=nil` since `device` is used in the `send` below? If it should be `nil`, would changing this to `ControlMessage.Unsubscribe(nil)` make that more clear?

Reply:
🤦🏼 should be ~= nil